File: L881-BoatsToSavePeople.java
// https://leetcode.com/problems/boats-to-save-people/
class Solution {
    public int numRescueBoats(int[] people, int limit) {
        int boats = 0, l = 0, r = people.length - 1;
        Arrays.sort(people);
        while (l <= r) {
            if (people[l] + people[r] <= limit) {
                boats += 1;
                l += 1;
            } else {
                boats += 1;
            }
            r -= 1;
        }
        return boats;
    }
}

File: L881-boats_to_save_people.py
# https://leetcode.com/problems/boats-to-save-people/
class Solution:
    def numRescueBoats(self, people: List[int], limit: int) -> int:
        people.sort()
        length = len(people)
        l, r = 0, length - 1
        boat = 0
        while (l <= r):
            if (people[l] + people[r] <= limit):
                boat += 1
                l += 1
            else:
                boat += 1
            r -= 1
        return boat
